### ISSUE TITLE
Any training tips?

### ISSUE THREAD

**Issue by conan1024hao** [CyberAgentAILab/layout-dm#29]
Hi, I am trying to apply LayoutDM to a custom dataset. I tried training using the default config (epoch=50, lr=5e-4, diffusion_step=100) but didn't get satisfactory results. My dataset contains irregular canvas including `H < W` wide ones.

Which training arguments are the most sensitive? Any training tips are welcome.

(お世話になっております。CAのインターン業務外の研究なのでここで質問させていただきます。)

**Comment by naoto0804** [comment]
Hi, thank you for your interest in our repo!

Hmm.. the parameters listed were not as sensitive as I tried, as long as a validation loss converges. If it converges very slowly, please try higher learning rate.

The most typical failure case I have encountered is when the learning rate decays too early. 
This is caused by a learning rate scheduler used in our code ([doc](https://pytorch.org/docs/stable/generated/torch.optim.lr_scheduler.ReduceLROnPlateau.html), [default params in our experiments](https://github.com/CyberAgentAILab/layout-dm/blob/main/src/trainer/trainer/config/scheduler/reduce_lr_on_plateau.yaml)), which adaptively decays the learning rate based on validation loss but it is sometimes too early; Please keep watching train/validation loss and if the learning rate reduces too early, you might set stricter criteria for the decay.

(あ，どうも〜！)

**Comment by naoto0804** [comment]
Additional note: we recently found that using a larger number of bins (e.g., 128) sometimes degrades the performance a lot;
